### DISCUSSION TITLE
GQL subscription implementation reference documentation

### DISCUSSION THREAD

**Discussion by Blemicek** [graphql-java-kickstart/graphql-java-servlet#265] [Q&A]
Hello, does exists any reference documentation for GQL subscription endpoint implementation? I've tried to understand it but I got completelly lost.

1. Is there any reason for custom subprotocol negotiation (which don't respect websockets RFC) instead of negotion provided by servlet containers (and JSR 356)?
2. Is it possible to intercept inbound and outbound messages (e.g. for logging or authorization reason which can be useful before GQL layer) or is it intendet to do it in intrumentations?
3. Is it possible to disable the fallback subscription protocol? (It seems no.) Why `WebSocketSubscriptionProtocolFactory` doesn't extend  `SubscriptionProtocolFactory` although `GraphQLWebsocketServlet` rely on it?
4. What is current status of subscription implementation? Is it considered production ready?

I apologize, if that sounds offensive, I just want to gather all the information to decide whether to use the library, implement my own solution, or try to improve yours.

**Comment by oliemansm** [comment]
@Blemicek The current websocket subscription implementation was done by one of the previous maintainers, and hasn't been touched/improved since then. It's basically the first working example, that is being used in production, but can definitely be improved. It sounds like you have quite some experience with websockets and good ideas that might benefit this implementation! Any help in this area would be great if you still are willing and have the time.
